Tests/TensorFlowTests/TensorGroupTests.swift

```diff
@@ -22,34 +22,129 @@ extension TensorDataType : Equatable {
     }
 }
 
-struct Empty : TensorGroup {}
+struct Empty : TensorGroup {
+    init() {}
+    init(handles: [_AnyTensorHandle]) {}
+    public var _tensorHandles: [_AnyTensorHandle] { [] }
+}
 
 struct Simple : TensorGroup, Equatable {
     var w, b: Tensor<Float>
+
+    init(w: Tensor<Float>, b: Tensor<Float>) {
+        self.w = w
+        self.b = b
+    }
+
+    init(handles: [_AnyTensorHandle]) {
+        precondition(handles.count == 2)
+        w = Tensor<Float>(handle: TensorHandle<Float>(handle: handles[0]))
+        b = Tensor<Float>(handle: TensorHandle<Float>(handle: handles[1]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] { [w.handle.handle, b.handle.handle] }
 }
 
 struct Mixed : TensorGroup, Equatable {
     // Mutable.
     var float: Tensor<Float>
     // Immutable.
     let int: Tensor<Int32>
+
+    init(float: Tensor<Float>, int: Tensor<Int32>) {
+        self.float = float
+        self.int = int
+    }
+
+    public init(handles: [_AnyTensorHandle]) {
+        precondition(handles.count == 2)
+        float = Tensor<Float>(handle: TensorHandle<Float>(handle: handles[0]))
+        int = Tensor<Int32>(handle: TensorHandle<Int32>(handle: handles[1]))
+    }
+
+    public var _tensorHandles: [_AnyTensorHandle] {
+        [float.handle.handle, int.handle.handle]
+    }
 }
 
 struct Nested : TensorGroup, Equatable {
     // Immutable.
     let simple: Simple
     // Mutable.
     var mixed: Mixed
+
+    init(simple: Simple, mixed: Mixed) {
```

Comment:
That is what I thought, but I get the following error, when I don't have these:
```
[1/2] Compiling TensorFlowTests TensorGroupTests.swift
/usr/local/google/home/bgogul/workspace/brain/s4tf/tensorflow-swift-apis/Tests/TensorFlowTests/TensorGroupTests.swift:173:22: error: cannot invoke initializer for type 'Simple' with an argument list of type '(w: Tensor<Float>, b: Tensor<Float>)'
        let simple = Simple(w: w, b: b)
                     ^
/usr/local/google/home/bgogul/workspace/brain/s4tf/tensorflow-swift-apis/Tests/TensorFlowTests/TensorGroupTests.swift:173:22: note: overloads for 'Simple' exist with these partially matching parameter lists: (_handles: C), (_owning: Optional<UnsafePointer<OpaquePointer>>\
)
        let simple = Simple(w: w, b: b)
                     ^
...
```

Is it because of the  the protocol initializers?

Reply:
Interesting. Yeah, that's probably why.